File: 2486-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        start = 0
        end = len(s) - 1
        moving = 0
        movingend = len(t) - 1
        while start <= end and moving <= movingend:
            if s[start] == t[moving]:
                moving += 1
                start += 1
            else:
                start += 1
        return len(t) - moving

File: 2486-append-characters-to-string-to-make-subsequence/append-characters-to-string-to-make-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var appendCharacters = function(s, t) {
  let start = 0;
  let end = s.length - 1;
  let moving = 0;
  let movingend = t.length - 1;
  while (start <= end && moving <= movingend) {
    if (s[start] == t[moving]) {
        moving++;
        start++;
    } else {
        start++;
    }
  }
  return t.length - moving;
};